app/main.py

```diff
@@ -1,13 +1,22 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors":
+            [format_linter_error(error) for error in errors if errors],
+            "path": file_path,
+            "status": "failed" if errors else "passed"
+            }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [format_single_linter_file(key, linter_report[key])
+            for key in linter_report.keys()
```

Comment:
and here you have to use descriptive and correct variable names. Rename your `key` variable
**Notice** if you iterate on linter_report it will return keys without specifying by method `.keys`

Reply:
that means that 
```python
for key in linter_report.keys()
```
is the same as
```
for key in linter_report
```
so you can make our code easier for reading